### ISSUE TITLE
default branch disappeared and project status is wrong afterwards

### ISSUE THREAD

**Issue by freakmoder** [amoerie/teamcity-theatre#151]
Hi,

I was setting up Teamcity Theatre and it was running fine a few days ago, but today I recognized that on one project, the default branch disappeared even though it was built at the same time as the other projects in the overview (all come from the same VCS and have the same build trigger configured). The second problem is that the project turned red now because one feature branch is failing even though the default branch is successful on Teamcity.

![image](https://user-images.githubusercontent.com/19290858/62035218-a8df0900-b1ef-11e9-9376-b89497830aa2.png)

**Comment by amoerie** [comment]
For some reason it thinks that branch is now the default one. Did you
change anything in Teamcity itself?

**Comment by freakmoder** [comment]
No, there was no change in teamcity regarding the default branches. It still shows me the branch dev as default branch.

**Comment by amoerie** [comment]
I would check the logs to see if it's TeamCity itself that's saying this branch is now the default branch.
Check your appsettings.json to see where it's writing log files. You might have to modify the MinimumLevel to Verbose and restart the application to see more information.

You can also call the TeamCity REST API yourself, use the 'GetBuildsOfBuildConfiguration' value from your appsettings.json and try it in Postman or any other REST API client you might know. The default value is `builds/?locator=branch:(default:any,policy:active_history_and_active_vcs_branches),running:any,count:{count},buildType:(id:{buildConfigurationId})&fields=count,build(id,buildTypeId,number,status,state,percentageComplete,branchName,defaultBranch,href,webUrl,running-info(percentageComplete,elapsedSeconds,estimatedTotalSeconds,currentStageText),queuedDate,startDate,finishDate)`

**Comment by freakmoder** [comment]
I need to wait if this case happens again. After the default branch was built again, the Theatre showed it properly. It normally happens when there are too many builds in branches and the default branch didn't got any commit for a couple of days.

**Comment by amoerie** [comment]
Ah this is actually a known "issue"/"feature".

For each tile, TeamCityTheatre will fetch the last 100 builds, and then group them by branch. If the last 100 builds don't contain a build for the default branch, you'll see the behavior you saw earlier.

A possible improvement could be that, when this happens, you would fetch the last default branch build separately. I try to keep the number of calls to TeamCity as low as possible, because we're already spamming it quite a lot.

**Comment by freakmoder** [comment]
I think the default branch should always be fetched. It looks irritating when the status of the project has a different behavior just because there were too many builds ongoing on teamcity. The default branch of a project should always be part of the list.
